script/upgrades/MU08/MU08Checks.sol

```diff
@@ -0,0 +1,164 @@
+
+    // MentoGovernance contracts:
+    timelockProxy = IGovernanceFactory(contracts.dependency("GovernanceFactory")).governanceTimelock();
+  }
+
+  function run() public {
+    console.log("\nStarting MU08 checks:");
+    prepare();
+
+    verifyTokenOwnership();
+    verifyMentoV2Ownership();
+    verifyMentoV1Ownership();
+  }
+
+  function verifyTokenOwnership() public {
+    console.log("\n== Verifying token proxy and implementation ownership: ==");
+    address[] memory tokenProxies = Arrays.addresses(cUSDProxy, cEURProxy, cBRLProxy, eXOFProxy, cKESProxy);
+    //address[] memory tokenProxies = Arrays.addresses(cUSDProxy, cEURProxy, cBRLProxy, eXOFProxy, cKESProxy, POSProxy);
+
+    for (uint256 i = 0; i < tokenProxies.length; i++) {
+      verifyProxyAndImplementationOwnership(tokenProxies[i]);
+    }
+    console.log("🤘🏼Token proxies and implementations ownership transferred to Mento Governance🤘🏼");
+  }
+
+  function verifyMentoV2Ownership() public {
+    console.log("\n== Verifying MentoV2 contract ownerships: ==");
+    address[] memory mentoV2Proxies = Arrays.addresses(brokerProxy, biPoolMangerProxy, reserveProxy);
+    for (uint256 i = 0; i < mentoV2Proxies.length; i++) {
+      verifyProxyAndImplementationOwnership(mentoV2Proxies[i]);
+    }
+    address[] memory mentoV2NonupgradeableContracts = Arrays.addresses(
+      breakerBox,
+      medianDeltaBreaker,
+      valueDeltaBreaker
+    );
+    console.log("Verifying MentoV2 nonupgradeable contract ownerships:");
+    for (uint256 i = 0; i < mentoV2NonupgradeableContracts.length; i++) {
+      verifyNonupgradeableContractsOwnership(mentoV2NonupgradeableContracts[i]);
+    }
+    console.log("🤘🏼MentoV2 contract ownerships transferred to Mento Governance🤘🏼");
+  }
+
+  function verifyMentoV1Ownership() public {
+    console.log("\n== Verifying MentoV1 contract ownerships: ==");
+    address[] memory mentoV1Proxies = Arrays.addresses(
+      exchangeProxy,
+      exchangeEURProxy,
+      exchangeBRLProxy,
+      grandaMentoProxy
+    );
+    for (uint256 i = 0; i < mentoV1Proxies.length; i++) {
+      verifyProxyAndImplementationOwnership(mentoV1Proxies[i]);
+    }
+    console.log("🤘🏼MentoV1 contract ownerships transferred to Mento Governance🤘🏼");
+  }
+
+  function verifyProxyAndImplementationOwnership(address proxy) internal {
+    address proxyOwner = IOwnableLite(proxy).owner();
+    require(proxyOwner == timelockProxy, "❗️❌ Proxy ownership not transferred to Mento Governance");
+    console.log("🟢 Proxy:[%s] ownership transferred to Mento Governance", proxy);
+
+    address implementation = IProxyLite(proxy)._getImplementation();
+    address implementationOwner = IOwnableLite(implementation).owner();
+    require(implementationOwner != address(0), "❗️❌ Implementation not owned by anybody");
+    if (implementationOwner != timelockProxy) {
+      console.log("🟡 Warning Implementation:[%s] ownership not transferred to Mento Governance 🟡 ", implementation);
```

Comment:
maybe change this to a require as well? as we shouldn't hit this

Reply:
we do with MentoV1 implementations